packages/views/order_edit.py

```diff
@@ -0,0 +1,191 @@
+        'partner': partner,
+        'rates': partner.rates.split(','),
+        'has_unsaved_progress': (new_packages_cookie or updated_packages_cookie) and not request.POST.get('save'),
+        'is_staff': is_staff(request.user)
+    }
+    context.update(string_data_lists_context())
+
+    response = render(request, 'packages/order_edit.html', context)
+
+    if updated_packages_cookie:
+        response.set_cookie(str(order.id)+'_updated_packages', updated_packages_cookie)
+    if new_packages_cookie:
+        response.set_cookie(str(order.id)+'_new_packages', new_packages_cookie)
+
+    if request.POST.get('save'):
+        if updated_packages_cookie:
+            updated_packages = update_saved_packages(updated_packages_cookie)
+            send_update_email(updated_packages)
+        if new_packages_cookie:
+            new_packages = create_saved_packages(new_packages_cookie)
+            send_new_email(new_packages)
+        response.delete_cookie(str(order_id)+'_updated_packages')
+        response.delete_cookie(str(order_id)+'_new_packages')
+
+    return response
+
+
+def json_to_packages(json):
+    return [des.object for des in serializers.deserialize('json', json)]
+
+
+def change_packages_status(package_id, update_type, order):
+    package = Package.objects.filter(id=package_id)
+
+    if update_type == 'revert':
+        package.update(status=package.get().prev_status())
+    elif update_type == 'update':
+        package.update(status=package.get().next_status())
+    elif update_type == 'update-all':
+        for inner in order.related_packages():
+            inner.as_query().update(status=inner.next_status())
+    else:
+        for inner in order.related_packages():
+            inner.as_query().update(status=inner.prev_status())
+
+def update_saved_packages(updated_packages_cookie):
+    packages = serializers.deserialize('json', updated_packages_cookie)
+
+    for package in packages:
+        actual_package = Package.objects.filter(id=package.id)
+        actual_package.update(origin=package.origin, destination=package.destination, rate=package.rate,
+                              full_name=package.full_name, phone_number=package.phone_number, notes=package.notes)
+
+    return packages
+
+
+def create_saved_packages(new_packages_cookie):
+    packages = serializers.deserialize('json', new_packages_cookie)
+
+    for package in packages:
+        package = package.object
+        Package.objects.create(order=package.order, origin=package.origin, destination=package.destination,
+                               rate=package.rate, full_name=package.full_name, phone_number=package.phone_number,
+                               notes=package.notes)
+
+    return packages
+
+
+def send_update_email(packages):
+    #TODO:
+    #
+    pass
+
+
+def send_new_email(packages):
+    #TODO:
+    #
+    pass
+
+
+def save_changes_to_cookies(request, cookies, order, updated_packages_cookie, new_packages_cookie):

```

Comment:
Can't you work on multiple packages at once? I tried to and it doesn't work. It should. Try creating two packages before doing `save`.

Reply:
YOU CAN! I am aware of this issue, that's why I mentioned it in the related issue #16 and that's why it hasn't been moved to "Review in progress" or resolved yet.

Reply:
So at the moment you can't. But it's intended for the future.